pandas_flavor/register.py

```diff
@@ -1,5 +1,85 @@
 from functools import wraps
-from pandas.api.extensions import register_series_accessor, register_dataframe_accessor
+from pandas.api.extensions import (
+    register_series_accessor,
+    register_dataframe_accessor,
+)
+import inspect
+from contextlib import nullcontext
+
+method_call_ctx_factory = None
+
+
+def handle_pandas_extention_call(method, method_signature, obj, args, kwargs):
+    """
+    This function is called when the user calls the registered method on pandas dataframe object.
+    pandas extention mechanism passes args and kwargs of original method call as it was applied to obj
+
+    Implementation uses global var *method_call_ctx_factory"
+
+    a) case when *method_call_ctx_factory* is None
+    In this case the implementation calls the registered method with unmodified args and kwargs and returns underlying method result.
+
+    b) case when *method_call_ctx_factory* is not None
+    In this case *method_call_ctx_factory* expected to refer to the function to create the context object. The context object will be used
+    to process inputs and outputs of *method* call. It is also possible that the context object method *handle_start_method_call*
+    will modify original args and kwargs before *method* call.
+
+    method_call_ctx_factory function signature: (method_name: str, args: list, kwargs: dict) -> MethodCallCtx
+
+    MethodCallCtx is an abstract class:
+    class MethodCallCtx(abc.ABC):
+        @abstractmethod
+        def __enter__(self) -> None: raise NotImplemented
+        @abstractmethod
+        def __exit__(self, exc_type, exc_value, traceback) -> None: raise NotImplemented
+        @abstractmethod
+        def handle_start_method_call(self, method_name: str, method_signature: inspect.Signature, method_args: list, method_kwargs: dict) -> tuple(list, dict): raise NotImplemented
+        @abstractmethod
+        def handle_end_method_call(self, ret: object) -> None: raise NotImplemented
```

Comment:
Minor change - switched to indented format to make it easier to read.

```suggestion
    MethodCallCtx is an abstract class:
    class MethodCallCtx(abc.ABC):

        @abstractmethod
        def __enter__(self) -> None: 
            raise NotImplemented

        @abstractmethod
        def __exit__(self, exc_type, exc_value, traceback) -> None: 
            raise NotImplemented

        @abstractmethod
        def handle_start_method_call(self, method_name: str, method_signature: inspect.Signature, method_args: list, method_kwargs: dict) -> tuple(list, dict): 
            raise NotImplemented

        @abstractmethod
        def handle_end_method_call(self, ret: object) -> None: 
            raise NotImplemented
```